lib/client/ipam_block_reader_writer.go

```diff
@@ -92,6 +93,17 @@ func (rw blockReaderWriter) claimNewAffineBlock(
 		}
 	}
 
+reqPools:
```

Comment:
Alternatively something like this might work? Haven't tried it though.

```go
// Build a map so we can lookup existing pools.
pm := map[string]bool{}
for _, ap := range allPools.Items {
  pm[ap.Metadata.CIDR.String()] = true
}

// Make sure each requested pool exists.
for _, rp := range requestedPools {
  if _, ok := pm[rp.String()]; !ok {
    // The requested pool doesn't exist.
    return nil, fmt.Errorf("The given pool doesn't exist")
  }
}
```